updater/lib/github_api/dependency_submission.rb

```diff
@@ -0,0 +1,197 @@
+# typed: strict
+# frozen_string_literal: true
+
+# This class provides a data object that can be submitted to a repository's dependency submission
+# REST API.
+#
+# See:
+#   https://docs.github.com/en/rest/dependency-graph/dependency-submission
+module GithubApi
+  class DependencySubmission
+    extend T::Sig
+
+    SNAPSHOT_VERSION = 0
+    SNAPSHOT_DETECTOR_NAME = "dependabot"
+    SNAPSHOT_DETECTOR_URL = "https://github.com/dependabot/dependabot-core"
+
+    sig { returns(String) }
+    attr_reader :job_id
+    sig { returns(String) }
+    attr_reader :ref
+    sig { returns(String) }
+    attr_reader :sha
+    sig { returns(String) }
+    attr_reader :directory
+    sig { returns(String) }
+    attr_reader :package_manager
+    sig { returns(T::Hash[String, T.untyped]) }
+    attr_reader :manifests
+
+    sig { params(job: Dependabot::Job, snapshot: Dependabot::DependencySnapshot).void }
+    def initialize(job:, snapshot:)
+      @job_id = T.let(job.id.to_s, String)
+      # TODO: Ensure that the branch is always set for analysis runs
+      #
+      # For purposes of the POC, we'll assume the default branch of `main`
+      # if this is nil, but this isn't a sustainable approach
```

Comment:
Note: we do not plan to address this during POC phase - an analysis-only job will need to be specified which has `source.branch` as a required argument.